homeassistant/components/deconz/device_automation.py

```diff
@@ -0,0 +1,171 @@
+
+CONF_DIM_DOWN_SHORT_PRESS = "remote_button_dim_down_short_press"
+CONF_DIM_DOWN_SHORT_RELEASE = "remote_button_dim_down_short_release"
+CONF_DIM_DOWN_LONG_PRESS = "remote_button_dim_down_long_press"
+CONF_DIM_DOWN_LONG_RELEASE = "remote_button_dim_down_long_release"
+
+CONF_TURN_OFF_SHORT_PRESS = "remote_button_turn_off_short_press"
+CONF_TURN_OFF_SHORT_RELEASE = "remote_button_turn_off_short_release"
+CONF_TURN_OFF_LONG_PRESS = "remote_button_turn_off_long_press"
+CONF_TURN_OFF_LONG_RELEASE = "remote_button_turn_off_long_release"
+
+CONF_LEFT_SHORT_PRESS = "remote_button_left_short_press"
+CONF_LEFT_SHORT_RELEASE = "remote_button_left_short_release"
+CONF_LEFT_LONG_PRESS = "remote_button_left_long_press"
+CONF_LEFT_LONG_RELEASE = "remote_button_left_long_release"
+
+CONF_RIGHT_SHORT_PRESS = "remote_button_right_short_press"
+CONF_RIGHT_SHORT_RELEASE = "remote_button_right_short_release"
+CONF_RIGHT_LONG_PRESS = "remote_button_right_long_press"
+CONF_RIGHT_LONG_RELEASE = "remote_button_right_long_release"
+
+
+HUE_DIMMER_REMOTE = {
+    CONF_TURN_ON_SHORT_PRESS: 1000,
+    CONF_TURN_ON_SHORT_RELEASE: 1002,
+    CONF_TURN_ON_LONG_PRESS: 1001,
+    CONF_TURN_ON_LONG_RELEASE: 1003,
+    CONF_DIM_UP_SHORT_PRESS: 2000,
+    CONF_DIM_UP_SHORT_RELEASE: 2002,
+    CONF_DIM_UP_LONG_PRESS: 2001,
+    CONF_DIM_UP_LONG_RELEASE: 2003,
+    CONF_DIM_DOWN_SHORT_PRESS: 3000,
+    CONF_DIM_DOWN_SHORT_RELEASE: 3002,
+    CONF_DIM_DOWN_LONG_PRESS: 3001,
+    CONF_DIM_DOWN_LONG_RELEASE: 3003,
+    CONF_TURN_OFF_SHORT_PRESS: 4000,
+    CONF_TURN_OFF_SHORT_RELEASE: 4002,
+    CONF_TURN_OFF_LONG_PRESS: 4001,
+    CONF_TURN_OFF_LONG_RELEASE: 4003,
+}
+
+TRADFRI_REMOTE = {
+    CONF_TURN_ON_SHORT_PRESS: 1002,
+    CONF_TURN_ON_LONG_PRESS: 1001,
+    CONF_DIM_UP_SHORT_RELEASE: 2002,
+    CONF_DIM_UP_LONG_PRESS: 2001,
+    CONF_DIM_UP_LONG_RELEASE: 2003,
+    CONF_DIM_DOWN_SHORT_RELEASE: 3002,
+    CONF_DIM_DOWN_LONG_PRESS: 3001,
+    CONF_DIM_DOWN_LONG_RELEASE: 3003,
+    CONF_LEFT_SHORT_RELEASE: 4002,
+    CONF_LEFT_LONG_PRESS: 4001,
+    CONF_LEFT_LONG_RELEASE: 4003,
+    CONF_RIGHT_SHORT_RELEASE: 5002,
+    CONF_RIGHT_LONG_PRESS: 5001,
+    CONF_RIGHT_LONG_RELEASE: 5003,
+}
+
+REMOTES = {"RWL021": HUE_DIMMER_REMOTE, "TRADFRI remote control": TRADFRI_REMOTE}
+
+TRIGGER_SCHEMA = vol.All(
+    vol.Schema(
+        {
+            vol.Required(CONF_DEVICE_ID): str,
+            vol.Required(CONF_DOMAIN): DOMAIN,
+            vol.Required(CONF_PLATFORM): "device",
+            vol.Required(CONF_TYPE): str,
+            vol.Required(CONF_UNIQUE_ID): str,
+        }
+    )
+)
+
+
+async def async_attach_trigger(hass, config, action, automation_info):
+    """Listen for state changes based on configuration."""
+    config = TRIGGER_SCHEMA(config)
+
+    device_registry = await hass.helpers.device_registry.async_get_registry()
+    device = device_registry.async_get(config[CONF_DEVICE_ID])
+
+    trigger = REMOTES[device.model][config[CONF_TYPE]]
```

Comment:
We should probably guard in case model or type are wrong.

Reply:
Because people will be able to create device automations via YAML too

Reply:
Well they can, but it will require a lot of work (device id and unique id etc.) so I wonder if it is relevant